src/rest/remote-api.ts

```diff
@@ -74,14 +83,25 @@ export interface IRemoteAPI {
     getAll<T = che.workspace.Workspace>(): Promise<T[]>;
     getAllByNamespace<T = che.workspace.Workspace>(namespace: string): Promise<T[]>;
     getById<T = che.workspace.Workspace>(workspaceKey: string): Promise<T>;
-    create(config: che.workspace.WorkspaceConfig, params: IResourceCreateQueryParams): Promise<any>;
+    /**
+     * Creates a workspace from config.
+     *
+     * @param devfile workspace devfile.
+     * @param params optional creating params.
+     */
+    create<T = che.workspace.Workspace>(devfile: che.workspace.devfile.Devfile, params: IResourceCreateParams | undefined): Promise<T>;
     update(workspaceId: string, workspace: che.workspace.Workspace): Promise<any>;
     delete(workspaceId: string): Promise<any>;
-    start(workspaceId: string): Promise<any>;
-    startTemporary(config: che.workspace.WorkspaceConfig): Promise<any>;
+    /**
+     * Starts a workspace.
+     *
+     * @param workspaceId a workspace ID.
+     * @param params resource query params.
+     */
+    start(workspaceId: string, params?: IResourceQueryParams): Promise<any>;
     stop(workspaceId: string): Promise<any>;
     getSettings<T = WorkspaceSettings>(): Promise<T>;
-    getFactory<T = che.factory.Factory>(factoryId: string): Promise<T>;
+    getFactoryResolver<T = che.workspace.devfile.Devfile>(url: string, headers?: { [name: string]: string }): Promise<T>;
```

Comment:
```
interface FactoryResolver {
    v: string;
    source: string;
    devfile: che.workspace.devfile.Devfile
}
...
   getFactoryResolver<T = FactoryResolver>(url: string, headers?: { [name: string]: string }): Promise<T>;
```

Reply:
So, the default type provided was incorrect. But still, this method returns object that is the result of resolving a factory URL on server side, isn't it?

Reply:
Yes, the default type provided was incorrect(interface). I added the right one.

Yes, this method returns an object that is the result of resolving a factory URL on the server-side. It can be just generated default devfile.

Reply:
The returned object is not a devfile and is not a factory.
It is a factory resolver object which includes information about the source of the generated devfile. We need this information on the client-side on the dashboard. That is why the name of this method is ```getFactoryResolver```